### ISSUE TITLE
Errors when building for Android with plugins that include Google Play Services as a framework

### ISSUE THREAD

**Issue by jkasten2** [Microsoft/cordova-docs#15]
When building a Cordova CLI 5.1.1 project in Visual Studio 2015 that has a plugin that includes Google Play Services as a framework the following error is shown in the build console:

```cmd
1>------ Rebuild All started: Project: BlankCordovaApp2, Configuration: Debug Android ------
1>  ANDROID_HOME=C:\Program Files (x86)\Android\android-sdk
1>  JAVA_HOME=C:\Program Files (x86)\Java\jdk1.7.0_80
1>  Running: c:\Users\Josh\Documents\Visual Studio 2015\Projects\BlankCordovaApp2\BlankCordovaApp2\platforms\android\gradlew clean -b c:\Users\Josh\Documents\Visual Studio 2015\Projects\BlankCordovaApp2\BlankCordovaApp2\platforms\android\build.gradle -Dorg.gradle.daemon=true
1>
1>  FAILURE: Build failed with an exception.
1>
1>  * What went wrong:
1>  A problem occurred configuring root project 'android'.
1>  > Could not resolve all dependencies for configuration ':_debugCompile'.
1>     > Could not find com.google.android.gms:play-services-gcm:7.3.0.
1>       Searched in the following locations:
1>           https://repo1.maven.org/maven2/com/google/android/gms/play-services-gcm/7.3.0/play-services-gcm-7.3.0.pom
1>           https://repo1.maven.org/maven2/com/google/android/gms/play-services-gcm/7.3.0/play-services-gcm-7.3.0.jar
1>           file:/C:/Program Files (x86)/Android/android-sdk/extras/android/m2repository/com/google/android/gms/play-services-gcm/7.3.0/play-services-gcm-7.3.0.pom
1>           file:/C:/Program Files (x86)/Android/android-sdk/extras/android/m2repository/com/google/android/gms/play-services-gcm/7.3.0/play-services-gcm-7.3.0.jar
1>       Required by:
1>           :android:unspecified
1>     > Could not find com.google.android.gms:play-services-ads:7.3.0.
1>       Searched in the following locations:
1>           https://repo1.maven.org/maven2/com/google/android/gms/play-services-ads/7.3.0/play-services-ads-7.3.0.pom
1>           https://repo1.maven.org/maven2/com/google/android/gms/play-services-ads/7.3.0/play-services-ads-7.3.0.jar
1>           file:/C:/Program Files (x86)/Android/android-sdk/extras/android/m2repository/com/google/android/gms/play-services-ads/7.3.0/play-services-ads-7.3.0.pom
1>           file:/C:/Program Files (x86)/Android/android-sdk/extras/android/m2repository/com/google/android/gms/play-services-ads/7.3.0/play-services-ads-7.3.0.jar
1>       Required by:
1>           :android:unspecified
1>
1>  * Try:
1>  Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
1>
1>  BUILD FAILED
1>
1>  Total time: 2.121 secs
1>
1>  c:\Users\Josh\Documents\Visual Studio 2015\Projects\BlankCordovaApp2\BlankCordovaApp2\platforms\android\cordova\node_modules\q\q.js:126
1>                      throw e;
1>                            ^
1>  Error code 1 for command: cmd with args: /s /c ""c:\Users\Josh\Documents\Visual Studio 2015\Projects\BlankCordovaApp2\BlankCordovaApp2\platforms\android\gradlew" clean -b "c:\Users\Josh\Documents\Visual Studio 2015\Projects\BlankCordovaApp2\BlankCordovaApp2\platforms\android\build.gradle" -Dorg.gradle.daemon=true"
1>C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\ApacheCordovaTools\vs-mda-targets\Microsoft.MDA.FileMirroring.targets(352,5): error MSB3073: The command "platforms\android\cordova\clean.bat" exited with code 1.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
```

The problem can be repdocuded by adding [OneSignal](https://github.com/one-signal/OneSignal-Cordova-SDK) as a custom plugin and setting the Cordova CLI to `5.1.1`.

If I build this project from the command line by running `cordova build android` it runs just fine. If I run `platforms\android\gradlew -b platforms\android\build.gradle -Dorg.gradle.daemon=true` from the command line I get the same error as VS.

OneSignal's plugin.xml framework includes:
```xml
<framework src="com.google.android.gms:play-services-gcm:7.3.0" />
<framework src="com.google.android.gms:play-services-ads:7.3.0" />
```

Please look into fixing this in Visual Studio or include any settings that are needed to to fix this issue.

Thanks.

**Comment by Chuxel** [comment]
This is not related to Visual Studio itself but appears to be a plugin problem with Cordova CLI 5.1.1.  Here's the repro steps:

npm install -g cordova@5.1.1
cordova create testproj
cd testproj
cordova plugin add https://github.com/one-signal/OneSignal-Cordova-SDK.git
cordova platform add android
cordova build android

I see the same errors.

**Comment by jkasten2** [comment]
Thanks for the quick reply! I was able to reproduce your issue in the command line after I ran `npm install -g cordova@5.1.1` from your repro steps. The root of the issue was I had 2 Android SDKs installed. One with the Google Repository installed and the other without. Some how running `npm install -g cordova@5.1.1` lined up the command line with the same Android SDK VS was using.

For future reference for anyone else who runs into a dependency error with play-services-gcm run the Android SDK Manager and install Extras>"Google Repository" and make sure it is up to date.
![image](https://cloud.githubusercontent.com/assets/645861/8887865/0b8eebf8-324a-11e5-8b9d-8aad9ed6f7fd.png)


Thanks.
